src/transformers/models/auto/auto_factory.py

```diff
@@ -469,6 +470,24 @@ def from_pretrained(cls, pretrained_model_name_or_path, *model_args, **kwargs):
         if token is not None:
             hub_kwargs["token"] = token
 
+        if is_peft_available():
+            revision = kwargs.get("revision", None)
+            subfolder = kwargs.get("subfolder", None)
+
+            maybe_adapter_path = find_adapter_config_file(
+                pretrained_model_name_or_path,
+                revision=revision,
+                use_auth_token=use_auth_token,
+                subfolder=subfolder,
+            )
+
+            if maybe_adapter_path is not None:
+                with open(maybe_adapter_path, "r") as f:
```

Comment:
There seems to be a minor inconsistency, above you use:

```python
with open(maybe_adapter_model_path, "r", encoding="utf-8") as f:
```

Reply:
No, you should leave the encoding for Windows support.

Reply:
OK good to know, will revert it

Reply:
Done in [a4a361d](https://github.com/huggingface/transformers/pull/25077/commits/a4a361d1d505c3df0bfb326183788bbccf1d5cf9)